src/config.ts

```diff
@@ -3,6 +3,95 @@ import * as env from './utils/env'
 // User mention and submission name
 interface RejectionParams { user: string, name: string }
 
+type LogLocation = 'public' | 'thread' | 'none'
+
+export interface RejectionTemplate {
+  // The unique key used to identify the template
+  key: string
+
+  // The value sent to Discord for the enumeration
+  // 'value' should be identical to 'key'
+  enumValue: { name: string, value: string }
+
+  // The pretty value, for logging
+  prettyValue: string
+
+  // The templating function to create the rejection message
+  execute: (params: RejectionParams) => string
+
+  // The function to decide where to send the log
+  location: () => LogLocation
```

Comment:
does this need to be a function? if it were an enum we could just make this a discriminated union

Reply:
I want the API to be flexible for future requirements, which may involve some logic etc in the predicate.